contracts/vault-managers/CDPManager01.sol

```diff
@@ -222,8 +252,9 @@ contract CDPManager01 is ReentrancyGuard {
 
     // decreases debt
     function _repay(address asset, address owner, uint usdpAmount) internal {
-        uint fee = vault.calculateFee(asset, owner, usdpAmount);
+        uint fee = vault.checkpointFee(asset, owner);
```

Comment:
All the fee is paid at once. That also means that the `owner` may be required to have a significant amount of USDP in addition to `usdpAmount`.
Alternatively, we may pay the fee first, up to `usdpAmount`. And `repay` the remaining `usdpAmount - fee`, if any.

Reply:
applied